File: app/src/main/java/com/blockeq/stellarwallet/models/NativeAssetAvailability.java
package com.blockeq.stellarwallet.models;

public interface NativeAssetAvailability extends AssetAvailability {
    /**
     *    Native ASSET
     *    (base amount 2) * 0.5
     *    (default is one) additional signers * 1
     *    trustlines 2 * 0.5
     *    open offer * 0.5
     *
     *
     *    posted for trade X
     */
    Long getBaseAmount();
    Long getAdditionalSigners();
    Long getTrustline();
    Long getOpenOffers();
}

File: app/src/main/java/com/blockeq/stellarwallet/models/NativeAssetAvailability.kt
package com.blockeq.stellarwallet.models

interface NativeAssetAvailability : AssetAvailability {
    val baseCount: Int
    val baseAmount: Float
    val additionalSignersCount: Int
    val additionalSignersAmount: Float
    val trustLinesCount: Int
    val trustLinesAmount: Float
    val openOffersCount: Int
    val openOffersAmount: Float
}